gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java

```diff
@@ -74,7 +65,92 @@ public BaseTmxMapLoader (FileHandleResolver resolver) {
+        String staggerAxis = root.getAttribute("staggeraxis", null);
+        String staggerIndex = root.getAttribute("staggerindex", null);
+        String mapBackgroundColor = root.getAttribute("backgroundcolor", null);
+
+        MapProperties mapProperties = map.getProperties();
+        if (mapOrientation != null) {
+            mapProperties.put("orientation", mapOrientation);
+        }
+        mapProperties.put("width", mapWidth);
+        mapProperties.put("height", mapHeight);
+        mapProperties.put("tilewidth", tileWidth);
+        mapProperties.put("tileheight", tileHeight);
+        mapProperties.put("hexsidelength", hexSideLength);
+        if (staggerAxis != null) {
+            mapProperties.put("staggeraxis", staggerAxis);
+        }
+        if (staggerIndex != null) {
+            mapProperties.put("staggerindex", staggerIndex);
+        }
+        if (mapBackgroundColor != null) {
+            mapProperties.put("backgroundcolor", mapBackgroundColor);
+        }
+        mapTileWidth = tileWidth;
+        mapTileHeight = tileHeight;
+        mapWidthInPixels = mapWidth * tileWidth;
+        mapHeightInPixels = mapHeight * tileHeight;
+
+        if (mapOrientation != null) {
+            if ("staggered".equals(mapOrientation)) {
+                if (mapHeight > 1) {
+                    mapWidthInPixels += tileWidth / 2;
+                    mapHeightInPixels = mapHeightInPixels / 2 + tileHeight / 2;
+                }
+            }
+        }
+
+        Element properties = root.getChildByName("properties");
+        if (properties != null) {
+            loadProperties(map.getProperties(), properties);
+        }
+
+        Array<Element> tilesets = root.getChildrenByName("tileset");
+        for (Element element : tilesets) {
+            loadTileSet(map, element, tmxFile);
+            root.removeChild(element);
+        }
+
+        for (int i = 0, j = root.getChildCount(); i < j; i++) {
+            Element element = root.getChild(i);
+            loadLayer(map, map.getLayers(), element, tmxFile);
+        }
+        return map;
+    }
+
+    protected abstract void loadTileSet(TiledMap map, Element element, FileHandle tmxFile);
+
+    protected void loadLayer(TiledMap map, MapLayers parentLayers, Element element, FileHandle tmxFile) {
+        String name = element.getName();
+        if (name.equals("group")) {
+            loadTileGroup(map, parentLayers, element, tmxFile);
+        } else if (name.equals("layer")) {
+            loadTileLayer(map, parentLayers, element);
+        } else if (name.equals("objectgroup")) {
+            loadObjectGroup(map, parentLayers, element);
+        } else if (name.equals("imagelayer")) {
+            loadImageLayer(map, parentLayers, element, tmxFile);
+        }
+    }
+
+	protected void loadTileGroup(TiledMap map, MapLayers parentLayers, Element element, FileHandle tmxFile) {
 		if (element.getName().equals("group")) {
```

Comment:
this test already be done in loadLayer ?

Reply:
I didn't change this, this actually already was that way when I moved it here.

Reply:
I found a way to get rid of the unwanted whitespace changes. I can edit the file here in github. I cannot get it to not do it in IntelliJ Idea, even though I got the correct formatter. I cannot get the autoformat for this stuff to stop.